phantomgames/Home/Categories/CategoryRepository.swift

```diff
@@ -14,11 +14,9 @@ typealias CategoryResult = (String, Result<[Game], APIError>)
 protocol CategoryRepositoryType: AnyObject {
     func fetchAPIImage(completion: @escaping (Result<[CategoryResult], APIError>) -> Void)
     func fetchCategory(endpoint: String, completion: @escaping (Result<[Game], APIError>) -> Void)
-
 }
 class CategoryRepository: CategoryRepositoryType {
     
-    
     func fetchAPIImage(completion: @escaping (Result<[CategoryResult], APIError>) -> Void) {
         
         let dispatchGroup = DispatchGroup()
```

Comment:
why is this needed?

Reply:
for my filter by category feature. Each category has its own endpoint and there's at least 6 categories so the calls are organised using a dispatch group

Reply:
as stated in my warning it part of concurrency. so I left the repo in a state where it has no errors and im hoping the concurrency workshop will give more clarity on developing the feature